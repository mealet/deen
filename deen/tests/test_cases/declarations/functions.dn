fn foo() i32 {
  return 123;
}

fn greeter(name: *char) {
  println!("{}", name);
}

fn _strange_functionNameTEST123() {
  println!("strange function name printed");
}

fn add_numbers(a: i32, b: i32) i32 {
  return a + b;
}

fn ptr_arg_function(pointer: *i32) {
  println!("{}", *pointer);
}

fn main() i32 {
  println!("{}", foo());

  _strange_functionNameTEST123();

  println!("{}", add_numbers(5, 5));

  let a = 555;
  ptr_arg_function(&a);

  return 0;
}
