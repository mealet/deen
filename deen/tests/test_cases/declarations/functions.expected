123
strange function name printed
10
555
